File: Code/Factory.Java/src/factory/secondversion/BaseFactory.java
package factory.secondversion;

import factory.entities.ConcreteThing;
import factory.interfaces.IFactory;
import factory.interfaces.IThing;

/**
 * Factory that creates objects of the ConcreteThing class
 */
public class BaseFactory implements IFactory {

	@Override
	public IThing createThing() {
		return new ConcreteThing();
	}

}

File: Code/Factory.dotNet/Factory.dotNet/SecondVersion/BaseFactory.cs
using Factory.Interfaces;
using Factory.Models;

namespace Factory.SecondVersion
{
    /// <summary>
    /// Factory that creates objects of the ConcreteThing class
    /// </summary>
    public class BaseFactory : IFactory
    {
        public virtual IThing CreateThing()
        {
            return new ConcreteThing();
        }
    }
}
